### PR TITLE
Increment bnd version to 4.1.0

### PR DESCRIPTION
**Pull request by fritze2**
- Changed all bnd 4.0.0 versions to 4.1.0.
- Changed some Import-Package headers to match the same behavior we had in 4.0.0.
- Service-Component names that had numbers following periods are not valid in 4.1.0, so I replaced the periods with underscores where needed.
- In cases where bnd.bnd files included cnf/resources/bnd/app-resources.bnd and the Bundle-SymbolicName had numbers after periods, I removed the include and added the Service-Component header into the bnd.bnd instead.
- Fixed whitespace formatting.

Before (with locally cached dependencies): 

    time ./gradlew cnf:initialize
        BUILD SUCCESSFUL in 30s
        6 actionable tasks: 5 executed, 1 from cache

        real	0m31.509s
        user	0m1.094s
        sys	0m0.665s
    
    time ./gradlew release --parallel
        BUILD SUCCESSFUL in 11m 2s
        5499 actionable tasks: 5265 executed, 230 from cache, 4 up-to-date

        real	11m4.081s
        user	0m8.790s
        sys	0m1.487s

After (with locally cached dependencies):

    time ./gradlew cnf:initialize
        BUILD SUCCESSFUL in 36s
        6 actionable tasks: 5 executed, 1 from cache

        real	0m36.669s
        user	0m1.165s
        sys	0m0.366s

    time ./gradlew release --parallel
        BUILD SUCCESSFUL in 10m 57s
        5499 actionable tasks: 5406 executed, 89 from cache, 4 up-to-date

        real	10m58.419s
        user	0m6.398s
        sys	0m1.287s

### CODE REVIEW

**Comment by fritze2** [comment]
#build

**Comment by fritze2** [comment]
#build

**Comment by fritze2** [comment]
#build

**Comment by fritze2** [comment]
#build

**Comment by fritze2** [comment]
#build

File: dev/com.ibm.ws.cdi.1.2.ejb/bnd.bnd [hunk 1]
```diff
@@ -21,9 +21,7 @@ Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.7))"
   com.ibm.ws.cdi.ejb.security.EjbSecurityContextStore,\
   com.ibm.ws.cdi.ejb.interceptor.liberty.JCDIHelperFactoryImpl
 
-
-Service-Component: \
- com.ibm.ws.cdi.1.2.ejb.deserializationClassProvider;\
+Service-Component: com.ibm.ws.cdi_1_2.ejb.deserializationClassProvider;\
```
Comment (tevans78) [inline dev/com.ibm.ws.cdi.1.2.ejb/bnd.bnd]: Why the change to underscores?

Comment (fritze2) [reply dev/com.ibm.ws.cdi.1.2.ejb/bnd.bnd]: The component fully qualified name validator now considers numbers after a period invalid. Would you like to use a different format? It could be `com.ibm.ws.cdi12.ejb.deserializationClassProvider`. For names that have bVersion, they currently look like `${Bundle-SymbolicName}_1_0_24.*`.

Comment (tevans78) [reply dev/com.ibm.ws.cdi.1.2.ejb/bnd.bnd]: no, just curious
